control-plane/agents/src/bin/core/node/service.rs

```diff
@@ -319,32 +332,29 @@ impl Service {
         client.list_blockdevices(request).await
     }
 
+    /// Cordon the specified node.
     async fn cordon(&self, id: NodeId, label: String) -> Result<Node, SvcError> {
-        let spec = self
-            .registry
-            .specs()
-            .cordon_node(&self.registry, &id, label)
-            .await?;
+        let mut guarded_node = self.specs().guarded_node(&id).await?;
+
+        let spec = guarded_node.cordon(&self.registry, label.clone()).await?;
```

Comment:
Do we need to clone label and spec, can't we just move them?

Reply:
fixed